kernel/src/process.rs

```diff
@@ -185,7 +185,8 @@ impl<'a> Process<'a> {
 
     pub fn setup_mpu(&self, mpu: &::platform::MPU) {
         let data_start = self.memory.as_ptr() as usize;
-        let data_len = 12;
+        let data_len = closest_power_of_two(self.memory.len() as u32)
```

Comment:
Nope, kernel does help you. If @nealjack is ok for now, I don't mind accounting for that before merging this pull request.